test/groovy/PiperTestBase.groovy

```diff
@@ -0,0 +1,51 @@
+import com.lesfurets.jenkins.unit.BasePipelineTest
+
+import static ProjectSource.projectSource
+import static com.lesfurets.jenkins.unit.global.lib.LibraryConfiguration.library
+
+import org.junit.Rule
+import org.junit.rules.TemporaryFolder
+
+public class PiperTestBase extends BasePipelineTest {
+
+    @Rule
+    public TemporaryFolder pipelineFolder = new TemporaryFolder()
+
+    private File pipeline
+
+    protected messages = [], shellCalls = []
+
+    protected final void _setUp() {
+
+        super.setUp()
+
+        messages.clear()
+        shellCalls.clear()
+
+        preparePiperLib()
+
+        helper.registerAllowedMethod('echo', [String], {s -> messages.add(s)} )
+        helper.registerAllowedMethod('sh', [String], {  s ->
+            shellCalls.add(s.replaceAll(/\s+/, " ").trim())
+        })
+
+        pipeline = pipelineFolder.newFile()
+    }
+
+    protected withPipeline(p) {
+        pipeline << p()
+        loadScript(pipeline.getAbsolutePath())
```

Comment:
This does not work on a windows machine. Suggest change to:
`loadScript(pipeline.toURI().getPath())`

Reply:
done